src/components/forms/FileInput/FilePreview.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useRef, useState } from 'react'
+import classnames from 'classnames'
+
+import { SPACER_GIF } from './constants'
+
+export const FilePreview = ({
+  imageId,
+  file,
+}: {
+  imageId: string
+  file: File
+}): React.ReactElement => {
+  const fileReaderRef = useRef<FileReader>(new FileReader())
+  const [isLoading, setIsLoading] = useState(true)
+  const [previewSrc, setPreviewSrc] = useState(SPACER_GIF)
+  const [showGenericPreview, setShowGenericPreview] = useState(false)
+
+  useEffect(() => {
+    fileReaderRef.current.onloadend = (): void => {
+      setIsLoading(false)
+      setPreviewSrc(fileReaderRef.current.result as string)
+    }
+
+    fileReaderRef.current.readAsDataURL(file)
+
+    return (): void => {
+      fileReaderRef.current.onloadend = null
+    }
+  }, [])
+
+  const { name } = file
+
+  const onImageError = (): void => {
+    setPreviewSrc(SPACER_GIF)
+    setShowGenericPreview(true)
+  }
+
+  const isPDF = name.indexOf('.pdf') > 0
+  const isWord = name.indexOf('.doc') > 0 || name.indexOf('.pages') > 0
+  const isVideo = name.indexOf('.mov') > 0 || name.indexOf('.mp4') > 0
+  const isExcel = name.indexOf('.xls') > 0 || name.indexOf('.numbers') > 0
+  const isGeneric = !isPDF && !isWord && !isVideo && !isExcel
+
+  const imageClasses = classnames('usa-file-input__preview-image', {
+    'is-loading': isLoading,
+    'usa-file-input__preview-image--pdf': showGenericPreview && isPDF,
+    'usa-file-input__preview-image--word': showGenericPreview && isWord,
+    'usa-file-input__preview-image--video': showGenericPreview && isVideo,
+    'usa-file-input__preview-image--excel': showGenericPreview && isExcel,
+    'usa-file-input__preview-image--generic': showGenericPreview && isGeneric,
+  })
+
+  return (
+    <div
+      data-testid="file-input-preview"
+      className="usa-file-input__preview"
+      aria-hidden="true">
+      <img
+        id={imageId}
+        data-testid="file-input-preview-image"
+        src={previewSrc}
+        alt=""
+        className={imageClasses}
+        onError={onImageError}
+      />
+      {name}
```

Comment:
![Screen Shot 2021-02-18 at 10 57 24 AM](https://user-images.githubusercontent.com/10750442/108392418-52be2600-71d8-11eb-8336-1dd62e48ec82.png)
Got into this weird state when I had a file (screenshot) which had a long name. I wonder if we want to put the name in a block element or something with a set min width.. or should we file as bug to uswds

Reply:
Ah, I believe this is a result of needing to update to USWDS 2.8.1 (this update included some changes to the File Input CSS classes). I get the same result if I was on another branch with 2.8.0, and view the FileInput without running `yarn install`

Reply:
ha! looks like github deployments is just on time for me over here 😄